File: frontend/src/types/notification.ts
export type NotificationType =
  | "ticket_created"
  | "ticket_assigned"
  | "status_changed"
  | "comment_added"
  | "sla_warning"
  | "sla_breached"
  | "ticket_resolved";

export interface Notification {
  id: string;
  type: NotificationType;
  title: string;
  body: string | null;
  ticket_id: string | null;
  is_read: boolean;
  read_at: string | null;
  created_at: string;
}

File: backend/app/models/notification.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, String, Text, DateTime, UUID, ForeignKey
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import BaseModel


class Notification(BaseModel):
    __tablename__ = "notifications"

    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False)
    ticket_id: Mapped[uuid.UUID | None] = mapped_column(UUID(as_uuid=True), ForeignKey("tickets.id"), nullable=True)
    type: Mapped[str] = mapped_column(String(50), nullable=False)
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    body: Mapped[str | None] = mapped_column(Text, nullable=True)
    is_read: Mapped[bool] = mapped_column(Boolean, default=False, nullable=False)
    read_at: Mapped[datetime | None] = mapped_column(DateTime(timezone=True), nullable=True)

    user: Mapped["User"] = relationship("User", foreign_keys=[user_id])
    ticket: Mapped["Ticket | None"] = relationship("Ticket", foreign_keys=[ticket_id])
